### ISSUE TITLE
someOf always picks the first item in each loop

### ISSUE THREAD

**Issue by baocancode** [oftherivier/fictional#32]
I have noticed that if running `fictional.someOf(input, [0, 2], ['a', 'b'])` multiple times, `['b']` will never be a result. Here's an example code:

```
const fictional = require("fictional")

for (let index = 0; index < 100; ++index) {
  console.log(fictional.someOf(index, [0, 2], ['a', 'b']))
}
```

If changing the code to `fictional.someOf(input, [0, 1], ['a', 'b'])` then the results are only `[]` or `['a']`.

After some debugging, this line in someOf.js always assigns 0: `chosenIndex = id % remainingLen` due to `id` is not a number

I think the correct code should be: `chosenIndex = id.mod(remainingLen)` (some similar changes are in https://github.com/oftherivier/fictional/pull/30)

Some other places that might have the same issue:

![image](https://user-images.githubusercontent.com/30614076/198061030-3bb62e5a-f8ef-4300-8772-631f7a1d93e0.png)

**Comment by justinvdm** [comment]
Hey @baocancode, you're totally right, `someOf` should've be changed to use `.mod` with the change in #30. I'll fix this, thank you!

**Comment by justinvdm** [comment]
Fixed in `0.7.7`, thank you!

**Comment by baocancode** [comment]
Thanks for the fix, just a FYI that most (or all) examples in README have changed now after #30.

For example, `word('id-1')` returns `Ṯamoki` instead of `Minanȯ`.
